pkg/apis/core/v1/defaults.go

```diff
@@ -318,6 +318,27 @@ func SetDefaults_HTTPGetAction(obj *v1.HTTPGetAction) {
 		obj.Scheme = v1.URISchemeHTTP
 	}
 }
+
+// SetDefaults_Namespace adds a default label for all namespaces
+func SetDefaults_Namespace(obj *v1.Namespace) {
+	// TODO, remove the feature gate in 1.22
+	// we cant SetDefaults for nameless namespaces (generateName),
+	// we don't need to overwrite the pre-existing labelMetadataName
+	// This code needs to be kept in sync with the implementation that exists
+	// in Namespace Canonicalize strategy (pkg/registry/core/namespace)
+
+	// note that this can result in many calls to feature enablement in some cases, but
+	// we assume that theres no real cost there.
+	if utilfeature.DefaultFeatureGate.Enabled(features.NamespaceDefaultLabelName) {
+		if len(obj.Name) > 0 && obj.Labels[v1.LabelMetadataName] != obj.Name {
```

Comment:
lookups on nil maps don't panic, they return the zero value for the map value type 

https://play.golang.org/p/wMOQBwQnqMW

since normally namespaces will already match, I was trying to avoid assignments in the normal case

Reply:
will keep the original, unless you think it's worth changing here :)

Reply:
ok, I've changed my mind here and I'm force putting the obj.Name on the defaulting without checking :) This follows the same approach from Canonicalize, I think it's good to keep them in sync.